samples/sample_definition_and_criteria.rst

```diff
@@ -0,0 +1,104 @@
+.. _definition_and_criteria:
+
+Sample Definition and Criteria
+##############################
+
+Definition
+==========
+
+A sample is a basic Zephyr application that provides a brief overview of a feature, subsystem,
+or module. For example, Kernel services, drivers, protocols, etc.
+Samples should be limited in scope and should focus only on demonstrating non-trivial or
+essential aspects of the subsystem(s) or module(s) being highlighted.
+
+Sample Criteria
+===============
+
+1. Samples must not be used as a testcase.
+++++++++++++++++++++++++++++++++++++++++++
+  * The primary purpose of a sample is to provide a reference to the user.
+  * Samples must not use Zephyr's testing framework.
+
+    * Must not use :kconfig:option:`CONFIG_ZTEST` or :kconfig:option:`CONFIG_ZTEST_NEW_API`.
+    * Must not use zasserts in samples.
+
+  * If a sample can provide output that can be verified, then output should be evaluated against
+    expected value to have some level of testing for the sample itself.
+  * Samples are optional.
+
+2. Samples should target multiple platforms and architectures.
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+  * Minimize the use of ``platform_allow`` and architecture filters as it limits the scope
+    of testing to the mentioned platforms and architectures. Reference: `Testcase Docs`_
+  * Make use of ``integration_platforms`` to limit the scope when there are timing or
+    resource constraints.
```

Comment:
These refer to twister YAML file properties, which aren't discussed until the next section. Can the order be reversed or a reference added?